### ISSUE TITLE
@ExampleProperty/value is getting ignored

### ISSUE THREAD

**Issue by napolev** [springfox/springfox#3037]
### What version of the library are you using?
```
2.9.2
```

### What kind of issue is this?
```
Bug
```

### Example Repository

https://github.com/napolev/springfox-swagger2-issue

### How to recreate the issue
```
$ git clone https://github.com/napolev/springfox-swagger2-issue
$ cd springfox-swagger2-issue
$ gradle bootRun
```

### Description of the issue

On the repository pointed above there is a very simple project which contains the following file:

**/src/main/java/com/example/demo/car/CarController.java**

![Preview](https://i.ibb.co/MMdnnZz/image-3.png)

The issue is that the [@ExampleProperty / value](https://github.com/napolev/springfox-swagger2-issue/blob/master/src/main/java/com/example/demo/car/CarController.java#L43) is getting ignored as you can see on the image below:

![Preview](https://i.ibb.co/jvTj6Sw/image-2.png)

Other than that, everything works fine. For example, for the following payload the response is correct:

![Preview](https://i.ibb.co/dWtz21g/image.png)

**Comment by ironijunior** [comment]
+1

**Comment by charlycou** [comment]
+1

**Comment by trevarmand** [comment]
+1

**Comment by dilipkrish** [comment]
@napolev thank you for a detailed issue. I'll look into this issue. Can you see what the json looks like for the endpoint? Im curious if this is an issue with swagger-ui

**Comment by denbard** [comment]
+1

**Comment by Maegner** [comment]
@dilipkrish seems like a UI issue since the examples actually get to the FE but are never displayed.
<img width="476" alt="Screenshot 2019-09-06 at 16 30 45" src="https://user-images.githubusercontent.com/17438742/64440388-c8f1ca00-d0c3-11e9-91b9-466f044e9f08.png">

**Comment by markewallace** [comment]
+1

**Comment by napolev** [comment]
Hi @dilipkrish, any update on this? Thanks!

**Comment by marciosindeaux** [comment]
+1

**Comment by megatz** [comment]
+1

**Comment by shantis** [comment]
Any update on this issue? The examples are not working in 2.9.2.

**Comment by tlg-265** [comment]
is this project discontinued maybe?

**Comment by shantis** [comment]
So examples will never work in springfox -- I wonder how other organizations used swagger with examples in 2.0. Any pointers would really help. Thanks!

**Comment by tobHai** [comment]
Like @Maegner said the x-examples are loaded properly however they are not displayed in the UI.
Any hints on how to solve this ? 
As far as I understand the index.html in swagger-ui folder loads a certain webjar version of the UI but how can a change be made to this webjar?
Is it possible to load another version of the webjar?

**Comment by vemonet** [comment]
Apparently `x-examples` are already a workaround for POST queries:
https://github.com/swagger-api/swagger-ui/issues/1980 

I did not have issue using `examples` for post queries with OAS3 (which should be normal, I don't see why we should use a different property label for POST queries) resolved by https://smart-api.info/

The main issue is that properties annotated with @ApiModelProperty doesn't appears in the model in the generated Swagger JSON file, but I guess this would be another issue!

**Comment by nikosvaggalis** [comment]
I discovered that springfox produces x-examples in the json description file which is not recognized by spring-ui.The right format that spring ui correctly renders is having the example as part of  the schema block of the field.Also called **example** not **x-examples** 

`"schema":{                  "type":"string",
		      "example" : {
                       "property":"this is a test"
   }}`

all libraries versions are 2.9.2.
Can someone please verify?
thanks

**Comment by mdfass** [comment]
Has anybody tried this with 3.0.0.-SNAPSHOT?  I have the same issue with 2.9.2

**Comment by indicasta** [comment]
Hi! @mdfass  I already tried with 3.0.0-SNAPSHOT and still the same issue with 2.9.2. No change about it!

**Comment by nikosvaggalis** [comment]
yes,I've tried with 3.0.0-SNAPSHOT and it's still the same.Can you try to do what I've suggested to see if it works for you?

**Comment by indicasta** [comment]
Hi! @nikosvaggalis I have tried what you suggested before about changing the `JSON` file  produced by `Springfox` , replacing the "x-examples" by:
```
"schema":{ "type":"string", "example" : { "property":"this is a test" }}
```
For this purpose I took the first example in this issue provided by @napolev, first of all I changed the `Springfox` version from `2.9.2` to `3.0.0-SNAPSHOT` and reduced the `JSON` file obtained with this last version, then I converted the `JSON` file to `YML` using the swagger editor. You can find the result in https://github.com/indicasta/springfox-swagger2-issue/blob/master/api-docs-reduced.yml. Finally with this reduction I got the expected result shown in the image below. 

![image](https://user-images.githubusercontent.com/48498723/78210991-1c869380-74ac-11ea-9396-615b374ffb6a.png)

Do you have any idea on how can I get that result by using `Springfox`  automatically?

Just in case here you have the repo I created for this purpose:

https://github.com/indicasta/springfox-swagger2-issue

Thanks!

**Comment by tlg-265** [comment]
thank you @indicasta for your clear feedback. I think we need to know what Springfox annotations to use on top of the controller method in order to let Springfox to do the proper conversion from Java to Swagger syntax. I think this is a very common use case.

**Comment by napolev** [comment]
Bingo! Finally on this commit I was able to find the way to make `Sprinfox Swagger` to auto populate the values on the `Swagger UI`:

https://github.com/napolev/springfox-swagger2-issue/commit/7e2591d07fc5d8e83282f01bd7e8ce5e2475c761

Here you have a screenshot of it:

![image](https://user-images.githubusercontent.com/27621741/78320964-644efe80-7530-11ea-806f-221063f2dd93.png)

Thank you guys for all your help including @indicasta.

**Comment by prabhushrikant** [comment]
> Has anybody tried this with 3.0.0.-SNAPSHOT? I have the same issue with 2.9.2

I tried with 3.0.0-SNAPSHOT and issue still persists

**Comment by prabhushrikant** [comment]
> Hi! @nikosvaggalis I have tried what you suggested before about changing the `JSON` file produced by `Springfox` , replacing the "x-examples" by:
> 
> ```
> "schema":{ "type":"string", "example" : { "property":"this is a test" }}
> ```
> 
> For this purpose I took the first example in this issue provided by @napolev, first of all I changed the `Springfox` version from `2.9.2` to `3.0.0-SNAPSHOT` and reduced the `JSON` file obtained with this last version, then I converted the `JSON` file to `YML` using the swagger editor. You can find the result in https://github.com/indicasta/springfox-swagger2-issue/blob/master/api-docs-reduced.yml. Finally with this reduction I got the expected result shown in the image below.
> 
> ![image](https://user-images.githubusercontent.com/48498723/78210991-1c869380-74ac-11ea-9396-615b374ffb6a.png)
> 
> Do you have any idea on how can I get that result by using `Springfox` automatically?
> 
> Just in case here you have the repo I created for this purpose:
> 
> https://github.com/indicasta/springfox-swagger2-issue
> 
> Thanks!

I confirm it works in editor.swagger.io if you manual add example section but springfox simply ignores it if it finds a reference model and tries to infer the schema from there. It doesn't work if your model going to have dynamic keys or hashmap as a property.

**Comment by dilipkrish** [comment]
This has be fixed via #2767
